tests/performance/test_performance_testing.py

```diff
@@ -97,44 +102,112 @@ def build_test_suite(
     return contract, suite
 
 
-def run_tests(test_suite: TestSuite, contract: ContractClass):
-    asyncio.run(_run_tests_inner(test_suite, contract))
-
+async def prepare_suite(
+    manager: SyncManager, test_suite: TestSuite, contract: ContractClass
+) -> Tuple[TestRunner, SharedTestsState, Optional[TestExecutionState]]:
+    tests_state = SharedTestsState(
+        test_collector_result=TestCollector.Result(test_suites=[test_suite]),
+        manager=manager,
+    )
+    runner = TestRunner(
+        shared_tests_state=tests_state,
+        include_paths=[],
+    )
 
-async def _run_tests_inner(test_suite: TestSuite, contract: ContractClass):
+    # pylint: disable=protected-access
+    execution_state = await runner._prepare_suite(
+        test_contract=contract,
+        test_suite=test_suite,
+    )
+    return runner, tests_state, execution_state
+
+
+def wait_for_completion(test_suite: TestSuite, tests_state: SharedTestsState):
+    tests_left = len(test_suite.test_case_names)
+    while tests_left:
+        test_result = tests_state.get_result()
+        if isinstance(test_result, PassedTestCase):
+            out = sys.stdout
+        else:
+            out = sys.stderr
+
+        print(test_result.format(), file=out)
```

Comment:
Is `print` needed here?

Reply:
yep, kind of. helpful with debugging and it doesn't really hurt

Reply:
actually scratch that. i will delete those